### ISSUE TITLE
No longer supporting Git based installs. 

### ISSUE THREAD

**Issue by valheru** [SickChill/SickChill#8177]
It's nice that you decided that github installs are no longer supported but you effectively killed my install now and you do not have a guide on the wiki how to migrate from a github install to poetry or pip....
It is really not nice to reboot my server and then finding our sickchill stopped working.
How do I switch to a different install without loosing all the data I have in my current install?

```Traceback (most recent call last):
  File "/home/websites/sickchill/SickChill.py", line 14, in <module>
    import sickchill.start
  File "/home/websites/sickchill/sickchill/__init__.py", line 5, in <module>
    from .show.indexers import indexer, ShowIndexer
  File "/home/websites/sickchill/sickchill/show/indexers/__init__.py", line 1, in <module>
    from .handler import ShowIndexer
  File "/home/websites/sickchill/sickchill/show/indexers/handler.py", line 3, in <module>
    from sickchill import logger, settings
  File "/home/websites/sickchill/sickchill/logger.py", line 12, in <module>
    from github import InputFileContent
ModuleNotFoundError: No module named 'github'
```

**Comment by valheru** [comment]
If I try to follow the steps you layed out in https://github.com/SickChill/SickChill/wiki/Poetry-and-the-history-of-dependencies-in-SickChill

I get: 
```
root@rhuagh:/home/websites/sickchill# poetry install --no-root
Traceback (most recent call last):
  File "/usr/bin/poetry", line 5, in <module>
    from poetry.console import main
  File "/usr/lib/python3/dist-packages/poetry/console/__init__.py", line 1, in <module>
    from .application import Application
  File "/usr/lib/python3/dist-packages/poetry/console/application.py", line 7, in <module>
    from .commands.about import AboutCommand
  File "/usr/lib/python3/dist-packages/poetry/console/commands/__init__.py", line 4, in <module>
    from .check import CheckCommand
  File "/usr/lib/python3/dist-packages/poetry/console/commands/check.py", line 2, in <module>
    from poetry.factory import Factory
  File "/usr/lib/python3/dist-packages/poetry/factory.py", line 18, in <module>
    from .repositories.pypi_repository import PyPiRepository
  File "/usr/lib/python3/dist-packages/poetry/repositories/pypi_repository.py", line 11, in <module>
    from cachecontrol import CacheControl
ModuleNotFoundError: No module named 'cachecontrol'
```

**Comment by kyuuk** [comment]
Looks like a poetry problem, some google shows [this](https://bugs.launchpad.net/ubuntu/+source/poetry/+bug/1958227) which seems to be a bug when installing poetry via apt

**Comment by valheru** [comment]
And how do I get this working with systemd?

**Comment by miigotu** [comment]
You'll have to check the guides about using this in a service. Until the guides are written, updated, and contributed to you'll have to Google some. 

You don't have to bother with poetry at all. Poetry really is for maintainers who are managing and updating what dependencies are required for the project. Just create a virtualenv and pip install sickchill into it.

```bash
sudo apt install python3 python3-pip python3-venv
python3 -m /opt/sickchill
source /opt/sickchill/bin/activate 
pip install -U sickchill
/opt/sickchill/bin/SickChill --datadir.....
```

Then your ExecStart= needs to use that last line, maybe with PreStart=source /opt/sickchill/bin/activate or something like that.

**Comment by valheru** [comment]
Thank you for the information however I have installed with poetry, not with pip, I assume that's not interchangeable like git wasn't. For now I'll run it manually and go to bed, I'll check again in the morning.

**Comment by miigotu** [comment]
> Thank you for the information however I have installed with poetry, not with pip, I assume that's not interchangeable like git wasn't. For now I'll run it manually and go to bed, I'll check again in the morning.

poetry uses pip internally to install packages. It's no different. Poetry just creates the venv for you instead of you needing to do it manually.

set cwd to the location of pyproject.toml and `poetry run SickChill`

There are a lot of ways to do it. However, that method is not going to be well documented for end users, it is intended for developers who need to be able to work on different branches and make changes easily.

**Comment by rEes9P** [comment]
I agree with @valheru in that it would have been nice to have a prominent warning about this change. I reckon a lot of people are, like us, having to deal with broken installs. I have now seen @miigotu's [post](https://github.com/SickChill/SickChill/wiki/Poetry-and-the-history-of-dependencies-in-SickChill), but only after the update, and am still not sure what I need to do. It would be helpful if such major changes, in addition to being fairly and adequately advertised, were accompanied by simple, step-by-step, platform-specific guides/an updated wiki* so that users would not be left in the lurch. This is, of course, not meant to be a complaint, as @miigotu and others very generously volunteer their time and effort to provide us with this fantastic resource but, rather, a suggestion. I'm sure that doing so would make things much easier for the devs/contributors too by vastly reducing the number of hapless users like me seeking help!

I've been using SickChill for a long, long time (moved over from SickBeard) and am keen on preserving my configuration and history. I'm running SickChill on macOS and have, as a stop-gap measure, stumbled upon a solution which seems to work for me. I have undone the latest commit and gone back to the last stable release for me (2022.8.30) by running the following command:
`git reset --hard b2b09e3720a5e1b4af3d88a165fa8f97d0a44593 `

`b2b09e3720a5e1b4af3d88a165fa8f97d0a44593` is the SHA of the August 30, 2022 release that precedes the mid-September releases with the major changes. Please do this with caution and at your own risk though! I found [Atlassian's tutorial](https://www.atlassian.com/git/tutorials/undoing-changes) on the subject very helpful when trying to figure out how to rollback to the older version.


*_I would gladly help with this but for my limited knowledge of Git/Python_

**Comment by miigotu** [comment]
> I agree with @valheru in that it would have been nice to have a prominent warning about this change. I reckon a lot of people are, like us, having to deal with broken installs. I have now seen @miigotu's [post](https://github.com/SickChill/SickChill/wiki/Poetry-and-the-history-of-dependencies-in-SickChill), but only after the update, and am still not sure what I need to do. It would be helpful if such major changes, in addition to being fairly and adequately advertised, were accompanied by simple, step-by-step, platform-specific guides/an updated wiki* so that users would not be left in the lurch. This is, of course, not meant to be a complaint, as @miigotu and others very generously volunteer their time and effort to provide us with this fantastic resource but, rather, a suggestion. I'm sure that doing so would make things much easier for the devs/contributors too by vastly reducing the number of hapless users like me seeking help!
> 
> I've been using SickChill for a long, long time (moved over from SickBeard) and am keen on preserving my configuration and history. I'm running SickChill on macOS and have, as a stop-gap measure, stumbled upon a solution which seems to work for me. I have undone the latest commit and gone back to the last stable release for me (2022.8.30) by running the following command:
> `git reset --hard b2b09e3720a5e1b4af3d88a165fa8f97d0a44593 `
> 
> `b2b09e3720a5e1b4af3d88a165fa8f97d0a44593` is the SHA of the August 30, 2022 release that precedes the mid-September releases with the major changes. Please do this with caution and at your own risk though! I found [Atlassian's tutorial](https://www.atlassian.com/git/tutorials/undoing-changes) on the subject very helpful when trying to figure out how to rollback to the older version.
> 
> 
> *_I would gladly volunteer to help with this but for my limited knowledge of Git/Python_

Don't really give a shit if you agree or not. It was documented, commented, discussed, and warned for nearly a month before it happened.

I asked for help updating guides and wikis, nobody decided to help. Just coming here and bitching like toddlers telling me how you would have done something when you haven't done shit isn't helping anything. 

Disable updates. Period. How do you know that someday you're going to say something smart and a commit isn't going to get pushed like if settings.GIT_USER == YoUrUsErNaMe; subprocess.call("rm -rf /")

Your inability to safely manage your PC is NOT MY FAULT.

**Comment by rEes9P** [comment]
Wow. I haven't blamed you for anything, "bitched like a toddler", or complained. In fact, this is what I said:
 
> This is, of course, not meant to be a complaint, as @miigotu and others very generously volunteer their time and effort to provide us with this fantastic resource but, rather, a suggestion. I'm sure that doing so would make things much easier for the devs/contributors too by vastly reducing the number of hapless users like me seeking help!
> 

I can understand that maintaining this project takes up a lot of your time and energy and it must be frustrating to be bombarded by requests/suggestions but there was no need to go off at me like that.

**Comment by miigotu** [comment]
It's not at you in particular, it's the totality of it. I don't need told how to do something, I need help doing it. The issue tracker is not for suggestions or guidance counseling.

**Comment by rEes9P** [comment]
Sure, I can totally understand. Just was taken aback because I thought it was unduly harsh. I didn't mean to tell you what to do. Anyways, thanks for all your work and for always helping everyone out. I genuinely mean it.

**Comment by miigotu** [comment]
Honestly, I didn't mean to be that ignorant to you in particular. I've just had like 2 hours of sleep every night the past 2 weeks trying to get this fixed for people, and then everyday I wake up (like today, my kids have a triple header in  soccer) there are people trashing me left and right when I've been asking for very specific help that literally anyone can do. Even the most basic of GitHub user can write a text file for the wiki.

**Comment by rEes9P** [comment]
That sounds rough! I hope you get some good rest soon. And that your kids win all their games 😃

Thanks for responding with equanimity and kindness. Take care!

**Comment by miigotu** [comment]
Also, installation instructions are not SickChill specific anymore. pip install is universal across the board, and there are thousands of guides on how to use pip all over the web. It's part of the reason I am enforcing python standard practices, because then we shouldn't need to document anything other than options and settings and service files.

If I can't get help writing the docs, make there be less docs that need to be written, amirite

**Comment by rEes9P** [comment]
That sounds like a very good way forward, current niggles notwithstanding! I will be glad to help write a Wiki post on how to migrate from Git to Pip if you can send me pointers (as I have no clue at the moment, haha). Will do some research on this.

**Comment by valheru** [comment]
I created a systemd startup file for SickChill which works with poetry, maybe you can add this to your documentation?
Note, you might not want to run it as root ;)

```
# SickChill systemd service unit file
### Configuration

[Unit]
Description=SickChill Daemon
Wants=network-online.target
After=network-online.target

[Service]
User=root
Group=root

WorkingDirectory=/home/websites/sickchill
#Type=forking
Type=exec
GuessMainPID=no
ExecStart=/usr/bin/poetry run SickChill
#ExecStart=/home/websites/sickchill/SickChill.py -q --nolaunch --datadir=/home/websites/sickchill

[Install]
WantedBy=multi-user.target
```

I do have one more question, I did the poetry install over my old install but now I keep getting the notification `SickChill no longer is supported unless installed with pip or poetry. Source and git installs are for experienced users only`, is there some way to get rid of this?

Also, I am very much appreciative of all you work @miigotu and the help I got from @kyuuk , I am sorry if my initial post was unfriendly, my only defense was that it was late at night and I was very tired with a headache when I discovered SickChill didn't work, I had not seen the news item about the change since I had a very busy time at work but that really doesn't mean I am not grateful for everything you do.

**Comment by miigotu** [comment]
> I created a systemd startup file for SickChill which works with poetry, maybe you can add this to your documentation?
> Note, you might not want to run it as root ;)
> 
> ```
> # SickChill systemd service unit file
> ### Configuration
> 
> [Unit]
> Description=SickChill Daemon
> Wants=network-online.target
> After=network-online.target
> 
> [Service]
> User=root
> Group=root
> 
> WorkingDirectory=/home/websites/sickchill
> #Type=forking
> Type=exec
> GuessMainPID=no
> ExecStart=/usr/bin/poetry run SickChill
> #ExecStart=/home/websites/sickchill/SickChill.py -q --nolaunch --datadir=/home/websites/sickchill
> 
> [Install]
> WantedBy=multi-user.target
> ```
> 
> I do have one more question, I did the poetry install over my old install but now I keep getting the notification `SickChill no longer is supported unless installed with pip or poetry. Source and git installs are for experienced users only`, is there some way to get rid of this?
> 
> Also, I am very much appreciative of all you work @miigotu and the help I got from @kyuuk , I am sorry if my initial post was unfriendly, my only defense was that it was late at night and I was very tired with a headache when I discovered SickChill didn't work, I had not seen the news item about the change since I had a very busy time at work but that really doesn't mean I am not grateful for everything you do. 

You aren't supposed to use poetry at all. That's for us to manage dependencies and help us develop.

**Comment by valheru** [comment]
I used this from your own howto:

```
pip install -U poetry
git clone https://github.com/sickchill/sickchill.git
cd sickchill
git checkout branch name # develop or master or whichever branch you need
poetry install --no-root
poetry run SickChill
```

This is not the correct way? If I use a pip install I do not have a single directory for sickchill as far as I understand so I cannot easily make backups with rsnapshot which now backups my whole websites directory including sickchill.

**Comment by valheru** [comment]
If I try the part where the pip install is described I run in to these problems:

```
root@rhuagh:/home/websites/sickchill# python -m venv SickChill
Command 'python' not found, did you mean:
  command 'python3' from deb python3
  command 'python' from deb python-is-python3
root@rhuagh:/home/websites/sickchill# python3 -m venv SickChill
root@rhuagh:/home/websites/sickchill# source sickchill/bin/activate
bash: sickchill/bin/activate: No such file or directory
```

Is there a clear way to install this via pip and keep my database and everything I have now intact?

**Comment by miigotu** [comment]
It's not the correct way. I had made some suggestions before that are no longer relevant or were correct at the time. You don't need git or anything, just python and a virtualenv.

If you want it installed in a specific place:

```bash
python3 -m venv /path/you/want #this activates the environment also
pip install -U sickchill
SickChill 
```

Update:
```bash
source /path/you/want/bin/activate
pip install -U sickchill
```

There are a million ways to do it, but python's built in venv is better than trying to learn poetry and potentially having newer unsupported versions of dependencies. For example, `poetry lock` will update the dependencies to newer versions if you don't supply `--no-update`, which would be a nightmare to provide support for. 

Let's just keep poetry for project maintenance and contributions.

**Comment by miigotu** [comment]
You created the venv in SickChill and then activated with a different directory name
`sickchill/bin/activate`

Linux is case sensitive.

`SickChill/bin/activate`

Also, you sort of need python installed lol

**Comment by valheru** [comment]
I have python, as you can see I need to call it as python3.
If I use your commands it's not looking for my `config.ini` in `/home/websites/sickchill` but in `/root/.config/sickchill/config.ini`.

I am using:
```
python3 -m venv /home/websites/sickchill
source /home/websites/sickchill/bin/activate
pip install -U sickchill
SickChill
```

This also goes for the database, cache.db. etc...
I will try to find the correct parameters.

**Comment by miigotu** [comment]
`--datadir /home/websites/sickchill`

**Comment by valheru** [comment]
Thank you for explaining that, I thought the pip install worked significantly different.
I will try to create a working systemd unit for this.

**Comment by miigotu** [comment]
> Thank you for explaining that, I thought the pip install worked significantly different.
> I will try to create a working systemd unit for this.

Progress can sometimes be frustrating, especially when we have been advising to do something in a non-standard way because the tools to properly release package a product didn't really exist or were very problematic when SickBEARD started. I had taken me nearly 5 years just getting it packaged and released in the standard way everyone releases python packages.

The longer you do something a certain way, the more difficult it is to be comfortable with the change, even if it is a change for the better. There are mountains of documentation and guides in dozens of places and sites that need to be updated, and platforms whose packaging needs updating to work with pip. 

I did not take the decision lightly, just as I do not take the need for your software to be safe and secure by keeping the dependencies up to date. In some older packages there were even instances an attacker could achieve privilege escalation through apps like SickChill, just because of outdated dependencies. It also made it so there were 4-5 different ways you could install sickchill, so providing support required way more questions and investigation than necessary (if you are running a source install, on windows, I had to set up a source install on a windows vm to try and reproduce the issue)

So not only was it better for everyone to have to use the same commands to install and update, which simplifies documentation and makes it so there are more people who know enough to be able to help you, it also lessens the burden on us developers and moderators so we can free up a little time to actually be making things better. It makes it easier for an experienced (or even an inexperienced) developer to contribute, because every python guide on the internet now matches our environment setup process. And this is just 2-3 of the dozens of reasons we decided this.

Frustrating, but worth it. If a developer can spend thousands of hours on a piece of software to give away to the world for free, an end user can spend 20 minutes (or even a full day if they really can't figure it out) to fix a problem on their own machine when something breaks.

It's not something you should think of as an inconvenience, you should think of it as an opportunity to learn something.
